OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -671,6 +672,167 @@ public static DreamValue NativeProc_clamp(DreamObject instance, DreamObject usr,
             return new DreamValue(0);
         }
 
+        [DreamProc("gradient")]
+        [DreamProcParameter("A", Type = DreamValueType.DreamObject)]
+        [DreamProcParameter("index", Type = DreamValueType.Float)]
+        public static DreamValue NativeProc_gradient(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            if (arguments.ArgumentCount <= 1) {
+                throw new IndexOutOfRangeException("Not enough arguments");
+            }
+
+            // We dont want keyword arguments screwing with this
+            DreamValue indx;
+
+            int colorSpace = 0;
+
+            List<DreamValue> gradientList;
+
+            if (arguments.GetArgument(0, "A").TryGetValueAsDreamList(out DreamList gradList)) {
+                gradientList = gradList.CreateCopy().GetValues();
```

Comment:
Creating a copy of the list is wasteful. It'd be better if this instead skipped over `"loop"` values in the `foreach` below.

Reply:
Then it will be interpreted as black

Reply:
Not if you skip over it? You could put this [here](https://github.com/OpenDreamProject/OpenDream/pull/837/files/cc650547589f6f969d32d3379e942630145d7282#diff-3de8438318b4a252c85ee8c5a1f174497034db870542953bb96764330f282f08R739).

```cs
if (colorStr == "loop")
    continue;
```